Data/ByteString/Internal.hs

```diff
@@ -648,6 +649,31 @@ concat = \bss0 -> goLen0 bss0 bss0
    concat [x] = x
  #-}
 
+-- | /O(log n)/ Repeats the given ByteString n times.
+times :: Integral a => a -> ByteString -> ByteString
+times n (BS fp len)
+  | n < 0 = error "stimes: non-negative multiplier expected"
+  | n == 0 = mempty
+  | len == 0 = mempty
+  | n == 1 = BS fp len
```

Comment:
Does returning `BS fp len` instead of a variable bound to the `BS` pattern result in additional allocations?

Reply:
Just checked with ddump-simpl and couldn't see any differences outside of different identifiers.
ddump-core-stats also returns the same amount of Terms, Types and Coercions, so I think it's optimized away. Wouldn't say no to assigning it to a name though if you believe it would be more readable

Reply:
The assembly generated with ddump-asm also looks essentially the same, up to renaming.

Reply:
Thanks for investigating! :+1: